src/os/bluestore/BlueStore.cc

```diff
@@ -1701,7 +1701,8 @@ void BlueStore::ExtentMap::reshard(Onode *o)
 	  for (const auto& sh : shards) {
 	    if (bstart < sh.offset && bend > sh.offset) {
 	      uint32_t blob_offset = sh.offset - bstart;
-	      if (b->get_blob().can_split_at(blob_offset)) {
+	      if (b->get_blob().can_split_at(blob_offset) &&
+		  blob_offset % min_alloc_size == 0) {
```

Comment:
And another concern - are there any means that ensure blob_offset to be aligned with min_alloc_size often enough. If no and blob_offset is pretty arbitrary then split blob functionality makes no sense...